### PR TITLE
Update alteration parser logics

### PR DESCRIPTION
**Pull request by zhx828**
There are more improvements we can add. But this is the one to fix a 500 error.

### CODE REVIEW

File: src/main/java/org/mskcc/oncokb/curation/service/MainService.java [hunk 1]
```diff
@@ -110,6 +140,7 @@ public void annotateAlteration(Alteration alteration) {
             alteration.setGenes(pcAlteration.getGenes());
         }
         alteration.setAlteration(pcAlteration.getAlteration());
+        alteration.setName(pcAlteration.getName());
```
Comment (bprize15) [inline src/main/java/org/mskcc/oncokb/curation/service/MainService.java:143]: Line 197 has duplicate logic with a check. Would remove one or the other depending how confident you are alteration name won't be null.

File: src/main/java/org/mskcc/oncokb/curation/service/MainService.java [hunk 2]
```diff
@@ -85,7 +88,31 @@ public Optional<Sequence> findSequenceByGene(ReferenceGenome referenceGenome, In
...
+                        }
+                        return gene;
+                    })
+                    .collect(Collectors.toSet())
+            );
+        }
+    }
+
     public void annotateAlteration(Alteration alteration) {
+        enrichQuery(alteration);
```
Comment (bprize15) [inline src/main/java/org/mskcc/oncokb/curation/service/MainService.java:114]: Logic on line 147 seems to do same thing. Do we need both?

File: src/main/java/org/mskcc/oncokb/curation/service/MainService.java (hunk 1 above)
Comment (zhx828) [reply src/main/java/org/mskcc/oncokb/curation/service/MainService.java:143]: That's to use protein change as name.

File: src/main/java/org/mskcc/oncokb/curation/service/MainService.java (hunk 2 above)
Comment (zhx828) [reply src/main/java/org/mskcc/oncokb/curation/service/MainService.java]: Good call. Updated the logics.

Comment (bprize15) [reply src/main/java/org/mskcc/oncokb/curation/service/MainService.java]: Sounds good otherwise LGTM

**Pull request merged**